### DISCUSSION TITLE
The `md:` class does not exist. If `md:` is a custom class, make sure it is defined within a `@layer` directive.

### DISCUSSION THREAD

**Discussion by rahulgore7** [tailwindlabs/tailwindcss#13006] [Help]
I am encountering this issue while developing this app with the tutorial. It also shows the same error for "lg:" classes

**Comment by alex-krasikau** [comment]
Hey @rahulgore7!

Could you share your tailwind config and show how you use the class in your project (a full line, or a file if possible)?

**Reply by SHB007** [reply]
I am facing the same issue.
My Tailwind.config.js
 /** @type {import('tailwindcss').Config} */
export default {
  content: [
    "./index.html",
    "./src/**/*.{js,ts,jsx,tsx}"
  ],
  theme: {
    container: {
      center: true,
      padding: "1rem",
    },
    extend: {
      fontFamily: {
        inter: ["Inter", "sans-serif"],
      },
      colors: {
        lwsGreen: "#00D991",
        deepDark: "#27292F",
        mediumDark: "#1E1F24",
        lighterDark: "#27292F",
      },
    },
  },
  // eslint-disable-next-line no-undef
  plugins: [require("@tailwindcss/typography")],
}



My index.css
@import url("https://fonts.googleapis.com/css2?family=Inter:wght@300;400;600;700&display=swap");
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
  body {
    @apply bg-deepDark font-inter text-gray-100;
  }
}

@layer components {
  * {
    font-family: "Inter", sans-serif;
    scroll-behavior: smooth;
  }

  ::-webkit-scrollbar {
    width: 10px;
  }

  ::-webkit-scrollbar-track {
    background: transparent;
  }

  ::-webkit-scrollbar-thumb {
    background: #888;
    border-radius: 10px;
  }

  ::-webkit-scrollbar-thumb:hover {
    background: #555;
  }

  /* Navbar */
  header {
    @apply text-white border-b border-slate-800 mb-4;
  }

  nav {
    @apply flex items-center flex-col md: flex-row justify-between md:p-4 py-2 md:py-6 gap-4 sticky top-0 min-h-[60px];
  }

  .avatar-img {
    @apply w-8 h-8 md: w-10 md:h-10 rounded-full grid place-items-center font-bold text-sm;
  }

  .avatar-img span {
    line-height: 0;
  }

  /* Blogs */
  .blog-card {
    @apply flex gap-4 flex-col md: flex-row bg-[#030317] p-3 border border-slate-800/40 rounded-lg transition-all hover:border-blue-900/50 hover:shadow-lg shadow-slate-200/50 cursor-pointer;
  }

  .blog-card>div {
    @apply flex-1;
  }

  .blog-thumb {
    @apply aspect-video object-cover rounded-md max-h-48;
  }

  /* Sidebar */
  .sidebar-card {
    @apply border border-slate-800/40 p-4 rounded-lg hover: border-slate-800/90;
  }

  .dot {
    @apply relative before: rounded-full pl-4 before:translate-y-[6px] before:absolute before:left-0 before:content-[''] before:w-1 before:h-1 before:bg-[#6C757D66];
  }

  /* Floating Actions */
  .floating-action {
    @apply fixed shadow-lg shadow-slate-900/60 bottom-10 left-1/2 -translate-x-1/2 bg-slate-800/80 backdrop-blur-md border border-slate-200/20 p-4 py-2 rounded-full;
  }

  .floating-action-menus {
    @apply flex gap-6;
  }

  .floating-action-menus li {
    @apply flex items-center gap-2 cursor-pointer hover: scale-105 hover:opacity-90 transition-all;
  }

  .floating-action-menus li span {
    @apply text-xs;
  }

  .floating-action-menus li img {
    @apply w-6 h-6;
  }

  /* Tags */
  .tags {
    @apply flex gap-2 mx-auto my-6 justify-center;
  }

  .tags li {
    @apply bg-slate-700 p-2 px-4 rounded-md text-sm cursor-pointer hover: bg-slate-800 hover:ring hover:ring-slate-500 transition-all;
  }

  /* createBlog */
  .createBlog {
    @apply w-8/12 mx-auto my-8;
  }

  .createBlog input {
    @apply w-full focus: ring-0 shadow-none border-none outline-none placeholder:text-3xl text-3xl pl-0 font-bold bg-transparent appearance-none focus:border-none focus:outline-none;
  }

  .createBlog textarea {
    @apply w-full focus: ring-0 border-none outline-none placeholder:text-lg text-lg pl-0 bg-transparent appearance-none focus:border-none focus:outline-none;
  }

  .createBlog #tags {
    @apply font-normal text-base placeholder: text-base text-slate-500 w-full placeholder:italic;
  }

  /* Action Modal */

  .action-modal-container {
    @apply absolute bg-slate-900 border border-slate-800 right-full top-6 min-w-[200px] divide-y divide-slate-800 py-1 rounded-md overflow-hidden;
  }

  .action-menu-item {
    @apply px-4 py-2 hover: bg-slate-800 flex w-full items-center gap-2 text-gray-100 transition-all;
  }
}

I am getting: 
<img width="659" alt="Screenshot_1" src="https://github.com/tailwindlabs/tailwindcss/assets/91208433/373d40d2-330b-42e7-8021-d8ae6b06931d">

**Reply by wongjn** [reply]
You'd need to remove the space before the `md:` prefix:

```diff
 nav {
-@apply flex items-center flex-col md: flex-row justify-between md:p-4 py-2 md:py-6 gap-4 sticky top-0 min-h-[60px];
+@apply flex items-center flex-col md:flex-row justify-between md:p-4 py-2 md:py-6 gap-4 sticky top-0 min-h-[60px];
 }

 .avatar-img {
-@apply w-8 h-8 md: w-10 md:h-10 rounded-full grid place-items-center font-bold text-sm;
+@apply w-8 h-8 md:w-10 md:h-10 rounded-full grid place-items-center font-bold text-sm;
 }

 .avatar-img span {
 line-height: 0;
 }

 /* Blogs */
 .blog-card {
-@apply flex gap-4 flex-col md: flex-row bg-[#030317] p-3 border border-slate-800/40 rounded-lg transition-all hover:border-blue-900/50 hover:shadow-lg shadow-slate-200/50 cursor-pointer;
+@apply flex gap-4 flex-col md:flex-row bg-[#030317] p-3 border border-slate-800/40 rounded-lg transition-all hover:border-blue-900/50 hover:shadow-lg shadow-slate-200/50 cursor-pointer;
 }
```
---
As an aside, Adam Wathan (creator of Tailwind) does seem to advocate avoiding `@apply`:
- https://twitter.com/adamwathan/status/1226511611592085504
- https://twitter.com/adamwathan/status/1559250403547652097

**Reply by SHB007** [reply]
Thank you very much. really a relief for me.

**Comment by rahulgore7** [comment]
Yes it was all because of the prettier extension. What worked for me was, I
just copied the code and did NOT hit the save button. Just keep it as it is
and it works fine.
